.github/workflows/regression.yml

```
@@ -75,34 +75,29 @@ jobs:
          ./run_playwright_tests.sh
         continue-on-error: true
 
-      - name: Download test report from S3
+      - name: Generate presigned URL for Playwright HTML report
+        id: presign
         run: |
-          mkdir -p ./playwright-report/html
-          aws s3 cp \
-            "s3://$AUTOMATED_TEST_REPORTS_BUCKET_NAME/staging/playwright-reports/latest/html/" \
-            playwright-report/html/ --recursive
+          DATETIME="${{ steps.run_tests.outputs.datetime }}"
+          ENVIRONMENT="${ENVIRONMENT}"
 
-          aws s3 cp \
-            "s3://$AUTOMATED_TEST_REPORTS_BUCKET_NAME/staging/playwright-reports/latest/results.json" \
-            playwright-report/results.json
+          REPORT_S3_PATH="s3://${AUTOMATED_TEST_REPORTS_BUCKET_NAME}/${ENVIRONMENT}/playwright-reports/${DATETIME}/html/index.html"
+          echo "Using report object: $REPORT_S3_PATH"
 
+          URL=$(aws s3 presign "$REPORT_S3_PATH" --expires-in 604800) # 7 days
+          echo "report_url=$URL" >> "$GITHUB_OUTPUT"
 
-      - name: Upload Playwright HTML report
-        uses: actions/upload-artifact@v4
-        with:
-          name: playwright-html-report-staging
-          path: playwright/playwright-report/html/
-
-      - name: Upload Playwright results JSON
-        uses: actions/upload-artifact@v4
+      - name: Publish Playwright report link
+        uses: actions/github-script@v7
         with:
-          name: playwright-results-json
-          path: playwright/playwright-report/results.json
-
-      - name: Fail job if tests failed
-        if: steps.run_tests.outcome != 'success'
```

How are we handing test failures now? Silent failures? Might consider `if: always()`, so that we still get a report. Thoughts?

---

Im going to revert that code back to the original. I think its possible to redesign and simplify much of the AWS code and it should result in smoother error handling. This will be done in a later spike ticket